### PR TITLE
OpenBB Handler & Agent Improvements

### PR DESCRIPTION
**Pull request by tmichaeldb**
## Description

This PR makes various improvements for OpenBB & Agents:
- Handle `LIMIT` and `ORDER_BY` correctly
- Use `duckdb` to handle functions in query targets (i.e. `AVG`, `MAX`, etc)
- Add a more detailed prompt/description for the text-to-SQL tool so agents can automatically determine which tables to query based on context
- Don't return 500 errors on the `/analysis/data` endpoint when analysis fails (can't properly handle dict return types from OpenBB). Long-term we should fix the `/analysis/data` endpoint, but the user should not see cryptic 500 errors on the GUI when this happens.
- Answer prompts in separate threads in `langchain_handler`

See the following demos to see how to use OpenBB with Chatbots:
- [MindsDB & OpenBB integration](https://www.loom.com/share/bc3e8a5fd013446c8f03fed69854be7e)
    - [SQL commands here](https://pastebin.com/8spNESHE) 
- [World News Knowledge Base](https://www.loom.com/share/859d9e672fce43e7b3f9212bb02a6fc1)
    - [SQL commands here](https://pastebin.com/gTCypTW9)
- [Creating & Publishing a Chatbot](https://www.loom.com/share/b674b13368e148fd91a85980f7d4b8c9)
    - [SQL commands here](https://pastebin.com/ep2QQsVY)
- [Chatbot Demo with text-to-SQL and knowledge bases](https://www.loom.com/share/42df3f26f29f403b9eaea9595a925974)

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [x]   Verification Steps: Experiment with text-to-SQL, OpenBB handler, and agents locally

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by DidierRLopes** [comment]
Legend.

Thank you for this @tmichaeldb

**Review by ZoranPandovski — APPROVED** [review]
LGTM. @TylerSandman We had some compatibilities issues between llama and langchain, is this version `langchain==0.0.347` compatible for both?

File: mindsdb/integrations/handlers/openbb_handler/openbb_tables.py [hunk 1]
```diff
@@ -43,6 +44,20 @@ def _get_params_from_conditions(self, conditions: List) -> Dict:
 
             return params
 
+        def _select_dataframe(self, query: ast.Select, df: pd.DataFrame) -> pd.DataFrame:
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/openbb_handler/openbb_tables.py:47]: looks like this function could be used instead: mindsdb.api.mysql.mysql_proxy.utilities.sql.query_df

**Comment by ea-rus** [comment]
Also in code of openbb handler: looks like self.obb is a global variable. It won't work on cloud (connection will be shared between users)

I think it should be replaced to something like:
```python
from openbb_core.app.static.app_factory import create_app
self.obb = create_app()
```

**Pull request merged**